sql/src/main/java/io/crate/operation/collect/collectors/OrderedCrateDocCollector.java

```diff
@@ -132,15 +135,28 @@ private int batchSize() {
 
     @Override
     public void doCollect() {
+        boolean isScoreRef = false;
+
         for (LuceneCollectorExpression<?> expression : expressions) {
             expression.startCollect(collectorContext);
             expression.setScorer(scorer);
         }
+
         searchContext.searcher().inStage(ContextIndexSearcher.Stage.MAIN_QUERY);
 
+        for (Symbol symbol : symbols) {
+            isScoreRef = ScoreReferenceDetector.detect(symbol);
+            if (isScoreRef)
+                break;
+        }
+
         TopFieldDocs topFieldDocs;
         try {
-            topFieldDocs = searchContext.searcher().search(searchContext.query(), batchSize(), sort);
+            if (isScoreRef) {
+                topFieldDocs = searchContext.searcher().search(searchContext.query(), null, batchSize(), sort, true, true);
```

Comment:
use this override for both calls and leave out condition
check if we really need doMaxScore